lib/Wikia/src/Service/User/Attributes/UserAttributes.php

```diff
@@ -91,9 +97,51 @@ private function setInMemcache( $userId, $attributes ) {
 	 * @param Attribute $attribute
 	 */
 	public function setAttribute( $userId, Attribute $attribute ) {
-		$this->setInService( $userId, $attribute );
+		$this->loadAttributes( $userId );
+
+		// If attribute value is null and default exists, set value to default
+		if ( is_null( $attribute->getValue() ) && isset( $this->defaultAttributes[$attribute->getName()] ) ) {
+			$attribute->setValue( $this->defaultAttributes[$attribute->getName()] );
+		}
+
 		$this->setInInstanceCache( $userId, $attribute );
-		$this->setInMemcache( $userId, $this->attributes[$userId] );
+	}
+
+	public function save( $userId ) {
+		$attributes = $this->loadAttributes( $userId );
+
+		// TODO When bulk updates are complete, convert this to a single request.
+		// Ticket: SOC-1482
+		$savedAttributes = [];
+		foreach( $attributes as $name => $value ) {
+			if ( $this->attributeShouldBeSaved( $name, $value ) ) {
+				$this->setInService( $userId, new Attribute( $name, $value ) );
+				$savedAttributes[$name] = $value;
+			} elseif ( $this->attributeShouldBeDeleted( $name, $value ) ) {
+				$this->deleteFromService( $userId, new Attribute( $name, $value ) );
+			}
+		}
+
+		$this->setInMemcache( $userId, $savedAttributes );
+	}
+
+	/**
+	 * Returns true if either is true:
+	 * 1.) No default for the attribute and the value is either not false or not null
+	 * 2.) Value is different than the default
+	 * @param $name
+	 * @param $value
+	 * @return bool
+	 */
+	private function attributeShouldBeSaved( $name, $value ) {
+		return (
+			( is_null( $this->defaultAttributes[$name] ) ) && !( $value === false || is_null( $value ) ) ||
```

Comment:
Not according to how MW has been saving attributes up to this point. I extracted the logic from [here|https://github.com/Wikia/app/blob/dev/includes/User.php#L5058] which performs the same check. I don't 100% understand why this has been the case, but wanted to keep the logic consistent.